docs/best_practices/python.md

```diff
@@ -0,0 +1,167 @@
+# Python best practices
+
+## Import
+
+### Avoid `from module_foo import symbol_bar`
+
+In general, it is a good practice to avoid the form `from foo import bar`
+because it introduces two distinct bindings (`bar` is distinct from `foo.bar`)
+and when the binding in one namespace changes, the binding in the other will
+not…
+
+That's also why this can interfere with the mocking.
+
+All in all, this should be avoided when unecessary.
+
+#### Rationale
+
+Reduce the likelihood of surprising behaviors and ease the mocking.
+
+#### Example
+
+``` python
+# Good
+import foo
+
+baz = foo.Bar()
+
+# Bad
+from foo import Bar
+
+baz = Bar()
+```
+
+#### References
+
+- [Idioms and Anti-Idioms in Python](https://docs.python.org/3.1/howto/doanddont.html#from-module-import-name1-name2%0A)
+- [unittest.mock documentation](https://docs.python.org/3.6/library/unittest.mock.html#where-to-patch)
+
+## Naming
+
+### Predicate functions
+
+Functions that return a Boolean value should have a name that starts with
+`has_`, `is_`, `was_`, `can_` or something similar that makes it clear that it's
```

Comment:
Not sure how a function could be a boolean? Or maybe were you anticipating the next line about `Boolean variable`?

Reply:
Indeed, it's confusing because I rewrote that part.

I'll reformulate as `that makes it clear that it returns a Boolean.`